### PR TITLE
Code analysis with static analysers 

### PR DESCRIPTION
**Pull request by HeroMaxPower**
## Description
Update of PR #2351, decided to start again.

Fixed multiple messages with static analysers, your IDE should be cleaner with Errors, Warnings and Information.

IDE0029 Null check can be simplified.
IDE0016 Null check can be simplified
IDE0019 Use pattern matching.
IDE0034 'default' expression can be simplified.
IDE0054 Use compound assignment.
IDE1005 Delegate invocation can be simplified.

S2971 Drop 'Where' and move the condition into the 'Count'.
S2971 Drop 'Where' and move the condition into the 'FirstOrDefault'.
S2971 Drop this useless call to 'ToList' or replace it by 'AsEnumerable' if you are using LINQ to Entities.
S1125 Remove the unnecessary Boolean literal(s).
S1116 Remove this empty statement.

### CODE REVIEW

**Comment by HeroMaxPower** [comment]
Hi @nohwnd or @Haplois is this something you could look at?

File: src/Microsoft.TestPlatform.Common/ExtensionFramework/Utilities/LazyExtension.cs [hunk 1]
```diff
@@ -78,17 +68,12 @@ public LazyExtension(TestPluginInformation pluginInfo, Type metadataType)
...
-                throw new ArgumentNullException("pluginInfo");
-            }
-
             if (metadata == null)
             {
                 throw new ArgumentNullException("metadata");
             }
 
-            this.testPluginInfo = pluginInfo;
+            this.testPluginInfo = pluginInfo ?? throw new ArgumentNullException("pluginInfo");
```
Comment (Haplois) [inline src/Microsoft.TestPlatform.Common/ExtensionFramework/Utilities/LazyExtension.cs:76]: Can you switch to `nameof` wherever applicable?

```suggestion
            this.testPluginInfo = pluginInfo ?? throw new ArgumentNullException(nameof(pluginInfo));
```

File: src/Microsoft.TestPlatform.Extensions.TrxLogger/Utility/TrxFileHelper.cs [hunk 2]
```diff
@@ -208,7 +208,7 @@ public string MakePathRelative(string path, string basePath)
             // and "\\dir1". We need to compare the first non-null token
             // to know if we've got differences.
             int i = 0;
-            for (i = 0; i < max && pathTokens[i].Length == 0 && basePathTokens[i].Length == 0; i++) ;
+            for (i = 0; i < max && pathTokens[i].Length == 0 && basePathTokens[i].Length == 0; i++)
```
Comment (Haplois) [inline src/Microsoft.TestPlatform.Extensions.TrxLogger/Utility/TrxFileHelper.cs:211]: This change incorrectly incorporates the following `if` block into `for`. 

```suggestion
            for (i = 0; i < max && pathTokens[i].Length == 0 && basePathTokens[i].Length == 0; i++)
            {
            }
```

File: test/vstest.console.UnitTests/TestPlatformHelpers/TestRequestManagerTests.cs [hunk 3]
```diff
@@ -307,12 +307,11 @@ public void DiscoverTestsShouldCollectMetrics()
...
-			Assert.IsTrue(actualRequestData.MetricsCollection.Metrics.TryGetValue(TelemetryDataConstants.TargetDevice, out targetDevice));
-			Assert.IsTrue(actualRequestData.MetricsCollection.Metrics.TryGetValue(TelemetryDataConstants.MaxCPUcount, out maxcount));
-			Assert.IsTrue(actualRequestData.MetricsCollection.Metrics.TryGetValue(TelemetryDataConstants.TargetPlatform, out targetPlatform));
-			Assert.IsTrue(actualRequestData.MetricsCollection.Metrics.TryGetValue(TelemetryDataConstants.DisableAppDomain, out disableAppDomain));
+            // Verify.
+            Assert.IsTrue(actualRequestData.MetricsCollection.Metrics.TryGetValue(TelemetryDataConstants.TargetDevice, out object targetDevice));
+            Assert.IsTrue(actualRequestData.MetricsCollection.Metrics.TryGetValue(TelemetryDataConstants.MaxCPUcount, out object maxcount));
+			Assert.IsTrue(actualRequestData.MetricsCollection.Metrics.TryGetValue(TelemetryDataConstants.TargetPlatform, out object targetPlatform));
+			Assert.IsTrue(actualRequestData.MetricsCollection.Metrics.TryGetValue(TelemetryDataConstants.DisableAppDomain, out object disableAppDomain));
```
Comment (Haplois) [inline test/vstest.console.UnitTests/TestPlatformHelpers/TestRequestManagerTests.cs:314]: This file is weirdly idented. Could you fix the identing as well?

File: src/DataCollectors/Microsoft.TestPlatform.Extensions.EventLogCollector/CollectorNameValueConfigurationManager.cs [hunk 4]
```diff
@@ -118,8 +118,7 @@ public CollectorNameValueConfigurationManager(XmlElement configurationElement)
                     return null;
                 }
 
-                string settingValue;
-                this.nameValuePairs.TryGetValue(name, out settingValue);
+                this.nameValuePairs.TryGetValue(name, out string settingValue);
```
Comment (Haplois) [inline src/DataCollectors/Microsoft.TestPlatform.Extensions.EventLogCollector/CollectorNameValueConfigurationManager.cs:121]: Most of these could be replaced by `out var` please do so where applicable.

```suggestion
                this.nameValuePairs.TryGetValue(name, out var settingValue);
```

**Review by Haplois — CHANGES_REQUESTED** [review]
@HeroMaxPower, thank you for the PR. Mostly nitpicking, but pointed out some problems as well.

File: src/DataCollectors/Microsoft.TestPlatform.Extensions.EventLogCollector/CollectorNameValueConfigurationManager.cs (hunk 4 above)
Comment (HeroMaxPower) [reply src/DataCollectors/Microsoft.TestPlatform.Extensions.EventLogCollector/CollectorNameValueConfigurationManager.cs]: Resolved where i found some

File: test/vstest.console.UnitTests/TestPlatformHelpers/TestRequestManagerTests.cs (hunk 3 above)
Comment (HeroMaxPower) [reply test/vstest.console.UnitTests/TestPlatformHelpers/TestRequestManagerTests.cs]: I think i've fixed indenting here

File: src/Microsoft.TestPlatform.Extensions.TrxLogger/Utility/TrxFileHelper.cs (hunk 2 above)
Comment (HeroMaxPower) [reply src/Microsoft.TestPlatform.Extensions.TrxLogger/Utility/TrxFileHelper.cs]: Fixed

File: src/Microsoft.TestPlatform.Common/ExtensionFramework/Utilities/LazyExtension.cs (hunk 1 above)
Comment (HeroMaxPower) [reply src/Microsoft.TestPlatform.Common/ExtensionFramework/Utilities/LazyExtension.cs]: Added where found

**Review by Haplois — APPROVED** [review]
LGTM. Thank you for the contribution, @HeroMaxPower.

**Pull request merged**